src/core/net/mdns.cpp

```diff
@@ -3634,7 +3724,10 @@ bool Core::RxMessage::ShouldSuppressKnownAnswer(const Question &aQuestion, const
 
 void Core::RxMessage::SendUnicastResponse(const AddressInfo &aUnicastDest)
 {
-    TxMessage response(GetInstance(), TxMessage::kUnicastResponse, aUnicastDest);
+    TxMessage response = mIsLegacyUnicast ? TxMessage(GetInstance(), TxMessage::kLegacyUnicastResponse, aUnicastDest,
+                                                      mMessagePtr, mQueryId)
+                                          : TxMessage(GetInstance(), TxMessage::kUnicastResponse, aUnicastDest);
```

Comment:
The reason I did this from constructor was to be able to populate the header with the received ```queryId```. If we are to do this, we might have to take the same approach as in ```TxMessage::Send()```, where header is read, populated with data, then wrote to ```mMsgPtr```. 
I could do something like ```TxMessage::Append**Question**From(OwnedPtr<Message> &aMessage)```, where I do have the queryId, as ```aMessage``` holds the header, and the question. This would be called from ```Core::RxMessage::SendUnicastResponse(const AddressInfo &aUnicastDest)``` if ```mIsLegacyUnicast``` is true, ```response.AppendQuestionFrom(mMessagePtr)```. Is this approach ok?

Reply:
I think adding `aQueryId` in constructor make perfect sense (so we can init the header properly).
But then we can add a method to copy question section from query to response? Does it make sense?